zgw/documenten-api/src/main/kotlin/com/ritense/documentenapi/client/DocumentenApiClient.kt

```diff
@@ -72,6 +77,78 @@ class DocumentenApiClient(
         return result
     }
 
+     fun storeDocumentInParts(
+        authentication: DocumentenApiAuthentication,
+        baseUrl: URI,
+        request: BestandsdelenRequest,
+        createDocumentResult: CreateDocumentResult,
+        bestandsnaam: String?
+    ): BestandsdelenResult {
+         requireNotNull(bestandsnaam) { "Bestandsnaam must be set otherwise uploading in bestanddelen will fail" }
+
+         // Inside the CreateDocumentResult there is an array of bestandsdelen.
+         // Each bestandsdeel needs to be sent separately
+         // So the documenten api determines the amount of chunks, not this application.
+         var response: BestandsdelenResult? = null
+
+         logger.info("Starting upload of file in {} chunks\n", createDocumentResult.bestandsdelen?.size)
+
+         for(bestandsdeel in createDocumentResult.bestandsdelen!!) {
+             logger.info("Sending chunk #{} for a size of {}",
+                 bestandsdeel.volgnummer,
+                 bestandsdeel.omvang)
+
+             val chunk = ByteArray(bestandsdeel.omvang)
+             val bytesRead = request.inhoud.read(chunk)
+
+             check (bytesRead == chunk.size) {
+                 "Failed to read all the bytes to upload. The expected omvang is larger than the bytes available."
+             }
+
+             val fileResource = object : ByteArrayResource(chunk) {
+                 override fun getFilename(): String {
+                     return bestandsnaam
+                 }
+             }
+
+             val body: MultiValueMap<String, Any> = LinkedMultiValueMap()
+             body.add("inhoud", fileResource)
+             body.add("lock", request.lock)
+
+             response = restClient(authentication)
+                 .put()
+                 .uri {
+                     ClientTools.baseUrlToBuilder(it, baseUrl)
+                         .path("bestandsdelen/{uuid}")
+                         .build(bestandsdeel.url.substring(bestandsdeel.url.lastIndexOf("/") + 1))
+                 }
+                 .contentType(MediaType.MULTIPART_FORM_DATA)
+                 .body(body)
+                 .retrieve()
+                 .body<BestandsdelenResult>()!!
+         }
+
+         check (request.inhoud.read() == -1) {
+             "Failed to upload the full file. The file is larger than the sum of the omvang of all bestandsdelen."
+         }
+
+         return response ?: throw IllegalStateException("Upload in chunks failed")
```

Comment:
This only returns the response of a single chunk upload. how valuable is that?

Reply:
The response is not used for anything I only used it for testing. Cleaning it up